ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1334,6 +1344,20 @@ public void dropTable(String tableName, boolean ifPurge) throws HiveException {
     dropTable(names[0], names[1], true, true, ifPurge);
   }
 
+  public void dropTable(Table table, boolean ifPurge) throws HiveException {
+    boolean tableWithSuffix = (HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_CREATE_TABLE_USE_SUFFIX)
+        || HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED))
+      && AcidUtils.isTransactionalTable(table)
+      && table.getSd().getLocation().matches(PATH_WITH_VERSION_ID);
+    
+    if (tableWithSuffix) {
+      long txnId = Optional.ofNullable(SessionState.get())
+        .map(ss -> ss.getTxnMgr().getCurrentTxnId()).orElse(0L);
+      table.getTTable().setTxnId(txnId);
+    }
```

Comment:
if the location is not PATH_WITH_VERSION_ID, we might have a situation were we are trying to do a softdelete, but could not do it. Do we need to handle this?

Reply:
we could have old tables that were created without this feature being enabled, and we still should support regular drops for them.

Reply:
Like above - we can always set the txnId, and use the flag to decide how to drop (soft/hard)